src/utils/api/reset-security-flow.service.ts

```diff
@@ -0,0 +1,85 @@
+import axios from '@lib/axios';
+import env from '@env';
+
+import { RESET_SECURITY_FLOW_TYPE } from '@enums/reset-security-flow-type';
+
+import type {
+  CreateResetPasswordData,
+  UpdateResetPasswordData,
+  CreateResetDeviceData,
+  UpdateResetDeviceData,
+} from '@type';
+
+/**
+ * Service for handling reset password and reset device flows.
+ *
+ * This is the service that is used by the reset password and reset device pages.
+ * This service is oriented on the server side `ResetSecurityFlowService`.
+ */
+class ResetSecurityFlowService {
```

Comment:
Classes or objects would have the advantage that we could define an interface or an abstract class that specifies what API methods should look like

Reply:
I really like the way the Stripe SDK is structured, broadly with everything namespaced by the resource type. I guess this is what you are saying around having an abstract class that defines the standard methods (get, list, delete, etc.)
```ts
stripe.customers.get()
stripe.invoices.get()
// etc.
```

The main reason I've left it as is for now is because I have a dream that this can all be [auto-generated](https://swagger.io/tools/swagger-codegen/) based on an OpenAPI spec (which is generated by our API controllers), then we can just do an import and not have to write any code ourselves

```ts
import beabee from @beabee/api

beabee.callouts.get(id) // or something!
```

So I'd like to look at this at the same time as looking into implementing an OpenAPI spec in general, as that will also auto-generate API docs as well.

Reply:
Yes, something like that. In this point of view, it makes sense to keep the old structure until we have a new uniform solution. Then we don't mix different approaches, which is then confusing.

We should also consider building a beabee client sdk as a separate NPM package, which others - who want to use the beabee api - can then also use